db/data.py

```diff
@@ -460,3 +462,41 @@ def get_summary_data(mbid, offset=0):
         pass
 
     return summary
+
+def find_md5_duplicates(md5):
+    """Check if data comes from an audio file with the same md5 (Uses data->md5_encoded)
+        
+    Args: 
+        md5: The md5_encoded from the low-level data to check
+    
+    Returns:
+        The retrieved mbid if the md5 already exists
+        None if the md5 is not found
+    """
+
+    query = text(
+            """SELECT lowlevel.mbid
+                 FROM lowlevel
+                 JOIN lowlevel_json USING (id)
+                WHERE lowlevel_json.data#> '{metadata,audio_properties,md5_encoded}' ? :md5""")
+    with db.engine.begin() as connection:
+        result = connection.execute(query, {"md5": md5})
+        row = result.fetchone()
+        if row:
+            # returns the retrieved mbid
+            return row[0]
+        else:
+            return None
+
+def generate_datasetitem_uuidv4():
+    """Generate a random UUIDv4 for dataset items that are not present in the DB as recordings
+    
+    Returns: 
+        itemuuid: The generated random UUIDv4 assigned to a specific dataset item
+    """
+    itemuuid = uuid4()
+    
+    if itemuuid:
+        return str(itemuuid)
+    else:
+        return None
```

Comment:
This function doesn't make much sense to me. Can just use `uuid4()`. It shouldn't fail. And even if it does we've got bigger problems and would like it to crash, not fail silently.